### ISSUE TITLE
DateTimeFormat: add 'skeleton' support 

### ISSUE THREAD

**Issue by jungshik** [tc39/ecma402#189]
Spun off from #108 and #119 . options.skeleton was discussed in #108, but I don't find any new bug filed. So, I'm doing it. If it's already filed and worked on, I'm sorry and a pointer to the existing bug would be appreciated. 

In Ecma 402 v1 back in early 2010's, there's a debate over verbose options ('month': short, 'month': 'numeric' , etc) vs LDML-style format specifiers (M, MM, MMM, MMMM).   Verbose options were what's agreed upon.   

Verbose options have some advantages, but they also have some cons; they're too verbose and sometimes hard to remember - devtools can help for sure. As shown in #119, it's hard to come up with succinct names for plethora of LDML format specifiers [1]  (field and values. especially values for timezone display format ). 

Quoting [@rxaviers 's summary](https://github.com/tc39/ecma402/issues/108#issuecomment-279976152) in #108, 

> options.skeleton - there's a big support for this option. It allows for "more advanced things", it is "almost ubiquitous for any internationalization platform". 

[1] I'm not necessarily saying that we have to support every single one of them.

**Comment by jungshik** [comment]
The existence of https://github.com/caridy/intl-datetimeformat-pattern ( despite its name, what this project does is more for this issue than for #190 ) is a good argument for this feature in a sense.

**Comment by caridy** [comment]
Agreed! I have done some more research on how to fulfill this features in user-land, and I can share the results soon.
